Defining DOCTEST_CONFIG_DISABLE globally causes linking errors

## Description
<!--
If your issue is a bugreport, this means describing what you did,
what did you want to happen and what actually did happen.

If your issue is a feature request, describe the feature and why do you
want it.
-->
I created a project based on the executable_dll_and_plugin example where I wanted to try doctest and see how it works when tests are implemented alongside library source code. Everything worked fine with my modifications to the example, until I wanted to try defining the DOCTEST_CONFIG_DISABLE globally (for production use case). I got the following linker error.
```
test_runner.obj : error LNK2019: unresolved external symbol "public: virtual __cdecl doctest::IReporter::~IReporter(void)" (??1IReporter@doctest@@UEAA@XZ) referenced in function "public: virtual void * __cdecl doctest::IReporter::`vector deleting destructor'(unsigned int)" (??_EIReporter@doctest@@UEAAPEAXI@Z) [C:\Users\Niketin\repos\doctest-example\build\dll\test_runner.vcxproj]
C:\Users\Niketin\repos\doctest-example\build\dll\Debug\test_runner.dll : fatal error LNK1120: 1 unresolved externals [C:\Users\Niketin\repos\doctest-example\build\dll\test_runner.vcxproj]
```
I wasn't able to solve this problem. I assume it is either a user error, or something is wrong in doctest, so I decided to open an issue here.

### Steps to reproduce
<!--
This is only relevant for bug reports, but if you do have one,
please provide a minimal set of steps to reproduce the problem.

Usually this means providing a small and self-contained code using doctest
and specifying compiler flags/tools used if relevant.
-->
1. Download and extract [doctest-example.zip](https://github.com/user-attachments/files/16970025/doctest-example.zip)
2. Build it with CMake:
   a. `cmake -B build .`
   b. `cmake --build build`
3. Build should complete fine. Listing tests show all 4 tests.
```
PS C:\Users\Niketin\repos\doctest_example> .\build\test\Debug\executable_dll_and_plugin.exe -ltc
[doctest] listing all test case names
===============================================================================
dll
dll utils
test_runner
executable
===============================================================================
[doctest] unskipped test cases passing the current filters: 4
```
4. Now, add `#define DOCTEST_CONFIG_DISABLE` at the beginning of each source file.
5. Build again. The error should come up.


### Extra information
<!--
Fill in any extra information that might be important for your issue.

If your issue is a bugreport, definitely fill out at least the following.
-->
* doctest version: **v2.4.11**
* Operating System: **Microsoft Windows 10 Pro Version 10.0.19045 Build 19045**
* Compiler+version: **MSVC 19.39.33523.0**

--- a/doctest/doctest.h
+++ b/doctest/doctest.h
@@ -6201,6 +6201,8 @@ DOCTEST_SUPPRESS_PRIVATE_WARNINGS_PUSH
 namespace doctest {
 #ifdef DOCTEST_CONFIG_DISABLE
 
+DOCTEST_DEFINE_INTERFACE(IReporter)
+
 int IReporter::get_num_active_contexts() {
     return 0;
 }

--- a/doctest/parts/private/reporter.cpp
+++ b/doctest/parts/private/reporter.cpp
@@ -8,6 +8,8 @@ DOCTEST_SUPPRESS_PRIVATE_WARNINGS_PUSH
 namespace doctest {
 #ifdef DOCTEST_CONFIG_DISABLE
 
+DOCTEST_DEFINE_INTERFACE(IReporter)
+
 int IReporter::get_num_active_contexts() {
     return 0;
 }

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -113,3 +113,4 @@ doctest_add_unit_test(
 doctest_add_unit_test(WITH_MAIN SOURCE regression/test.630.cpp)
 doctest_add_unit_test(WITH_MAIN SOURCE regression/test.651.cpp)
 doctest_add_unit_test(WITH_MAIN SOURCE regression/test.786.cpp)
+doctest_add_unit_test(          SOURCE regression/test.873.cpp)

--- a/tests/regression/test.873.cpp
+++ b/tests/regression/test.873.cpp
@@ -1,0 +1,38 @@
+#define DOCTEST_CONFIG_DISABLE
+#define DOCTEST_CONFIG_IMPLEMENT
+#include <doctest/doctest.h>
+
+namespace {
+
+DOCTEST_MSVC_SUPPRESS_WARNING_PUSH
+DOCTEST_MSVC_SUPPRESS_WARNING(4625)
+DOCTEST_MSVC_SUPPRESS_WARNING(5026)
+DOCTEST_MSVC_SUPPRESS_WARNING(4626)
+DOCTEST_MSVC_SUPPRESS_WARNING(5027)
+class reporter : public doctest::IReporter {
+public:
+    inline void report_query(const doctest::QueryData &) override { }
+    inline void test_run_start() override { }
+    inline void test_run_end(const doctest::TestRunStats &) override { }
+    inline void test_case_start(const doctest::TestCaseData &) override { }
+    inline void test_case_reenter(const doctest::TestCaseData &) override { }
+    inline void test_case_end(const doctest::CurrentTestCaseStats &) override { }
+    inline void test_case_exception(const doctest::TestCaseException &) override { }
+    inline void subcase_start(const doctest::SubcaseSignature &) override { }
+    inline void subcase_end() override { }
+    inline void log_assert(const doctest::AssertData &) override { }
+    inline void log_message(const doctest::MessageData &) override { }
+    inline void test_case_skipped(const doctest::TestCaseData &) override { }
+};
+DOCTEST_MSVC_SUPPRESS_WARNING_POP
+
+} // namespace
+
+int main() {
+  reporter r;
+  static_cast<void>(r);
+
+  // Required to fix issues with xcode optimizing away `main`
+  volatile int result = 0;
+  return result;
+}

